echo-pubsub-aws/src/test/java/com/netflix/spinnaker/echo/pubsub/aws/AmazonSQSSubscriberTest.java

```diff
@@ -0,0 +1,133 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+public class AmazonSQSSubscriberTest {
+
+  @Mock
+  private AmazonSNS amazonSNS;
+
+  @Mock
+  private AmazonSQS amazonSQS;
+
+  @Mock
+  private PubsubMessageHandler pubsubMessageHandler;
+
+  @Mock
+  private Registry registry;
+
+  @Mock
+  private Logger log;
+
+  private ARN queueARN = new ARN("arn:aws:sqs:us-west-2:100:queueName");
+  private ARN topicARN = new ARN("arn:aws:sns:us-west-2:100:topicName");
+
+  private AmazonPubsubProperties.AmazonPubsubSubscription subscription =
+    new AmazonPubsubProperties.AmazonPubsubSubscription(
+      "aws_events", topicARN.getArn(), queueARN.getArn(), "", null,
+      null, 3600);
+
+  private ObjectMapper objectMapper = EchoObjectMapper.getInstance();
+
+  private SQSSubscriber subject;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
+    subject = new SQSSubscriber(
+      objectMapper,
+      subscription,
+      pubsubMessageHandler,
+      amazonSNS,
+      amazonSQS,
+      () -> true,
+      registry
+    );
+  }
+
+  @Test
+  void shouldUnmarshalSNSNotificationMessage() throws JsonProcessingException {
+    // Given
+    String payload = "{\n" +
+      "  \"Records\": [\n" +
+      "    {\n" +
+      "      \"eventVersion\": \"2.0\",\n" +
+      "      \"eventSource\": \"aws:s3\",\n" +
+      "      \"awsRegion\": \"us-west-2\",\n" +
+      "      \"eventName\": \"ObjectCreated:Put\",\n" +
+      "      \"s3\": {\n" +
+      "        \"s3SchemaVersion\": \"1.0\",\n" +
+      "        \"configurationId\": \"prestaging_front50_events\",\n" +
+      "        \"bucket\": {\n" +
+      "          \"name\": \"us-west-2.spinnaker-prod\",\n" +
+      "          \"ownerIdentity\": {\n" +
+      "            \"principalId\": \"A2TW6LBRCW9VEM\"\n" +
+      "          },\n" +
+      "          \"arn\": \"arn:aws:s3:::us-west-2.spinnaker-prod\"\n" +
+      "        },\n" +
+      "        \"object\": {\n" +
+      "          \"key\": \"prestaging/front50/pipelines/31ef9c67-1d67-474f-a653-ac4b94c90817/pipeline-metadata.json\",\n" +
+      "          \"versionId\": \"8eyu4_RfV8EUqTnClhkKfLK5V4El_mIW\"\n" +
+      "        }\n" +
+      "      }\n" +
+      "    }\n" +
+      "  ]\n" +
+      "}";
+
+    NotificationMessageWrapper notificationMessage = new NotificationMessageWrapper(
+      "Notification",
+      "4444-ffff",
+      "arn:aws:sns:us-west-2:100:topicName",
+      "Amazon S3 Notification",
+      payload,
+      Collections.emptyMap()
+    );
+
+    String snsMessage = objectMapper.writeValueAsString(notificationMessage);
+
+    // When
+    String result = subject.unmarshalMessageBody(snsMessage);
+
+    // Then
+    verify(log, never()).error(anyString(), anyString(), any(IOException.class));
```

Comment:
This doesn't look correct.  The groovy code uses `subject.log`

Reply:
when we are using subject.log in `verify(subject.log, never()).error(anyString(), anyString(), any(IOException.class));`
we got below error:
`Argument passed to verify() is of type Logger and is not a mock!`